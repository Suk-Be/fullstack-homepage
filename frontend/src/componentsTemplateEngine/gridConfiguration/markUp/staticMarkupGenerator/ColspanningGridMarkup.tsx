import ColumnsSpanningGrid from '@/componentsTemplateEngine/presetRenderExamples/ColspanningGrid';
import colSpanningGridTagsToText from '@/utils/templateEngine/parseHtmlToText/colSpanningGridTagsToText';
import { ComponentPropsWithoutRef, FC } from 'react';
import { twMerge } from 'tailwind-merge';
import GridMarkupWrapper from '../generatorElements/GridMarkupWrapper';

interface ColspanningGridMarkupProps extends ComponentPropsWithoutRef<'div'> {
    hasCol: 'col-span-2';
}

const ColspanningGridMarkup: FC<ColspanningGridMarkupProps> = ({ hasCol, className }) => {
    return (
        <GridMarkupWrapper
            markupComponent={<ColumnsSpanningGrid className={hasCol} />}
            childrenTagsArr={colSpanningGridTagsToText}
            className={twMerge(className)} // Pass the className to the wrapper
        />
    );
};

export default ColspanningGridMarkup;
