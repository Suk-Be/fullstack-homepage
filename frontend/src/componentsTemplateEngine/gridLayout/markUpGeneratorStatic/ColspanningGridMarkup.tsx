import ColumnsSpanningGrid from '../../../componentsTemplateEngine/gridLayout/layoutPresets/ColspanningGrid';
import ChildrenTags from '../../../componentsTemplateEngine/gridLayout/markUpElements/ChildrenTags';
import ParentTag from '../../../componentsTemplateEngine/gridLayout/markUpElements/ParentTag';
import colSpanningGridTagsToText from '../../../utils/templateEngine/parseHtmlToText/colSpanningGridTagsToText';
import { ComponentPropsWithoutRef, FC } from 'react';
import { twMerge } from 'tailwind-merge';

interface ColspanningGridMarkupProps extends ComponentPropsWithoutRef<'div'> {
    hasCol: 'col-span-2';
}

const ColspanningGridMarkup: FC<ColspanningGridMarkupProps> = ({ hasCol, className }) => {
    return (
        <div className={twMerge('grid grid-cols-1 p-4', className)}>
            <div className="bg-gray-dark text-green p-6 rounded-xl">
                <ParentTag
                    isOpeningTag={true}
                    Component={<ColumnsSpanningGrid className={hasCol} />}
                />
                <ChildrenTags arr={colSpanningGridTagsToText} />
                <ParentTag
                    isClosingTag={true}
                    Component={<ColumnsSpanningGrid className={hasCol} />}
                />
            </div>
        </div>
    );
};

export default ColspanningGridMarkup;
