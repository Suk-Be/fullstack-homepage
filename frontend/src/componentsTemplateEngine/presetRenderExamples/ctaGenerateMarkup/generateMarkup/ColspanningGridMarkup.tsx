import ColumnsSpanningGrid from '@/componentsTemplateEngine/presetRenderExamples/grids/ColspanningGrid';
import colSpanningGridItemTagsToText from '@/utils/templateEngine/parseHtmlToText/colSpanningGridItemTagsToText';
import { ComponentPropsWithoutRef, FC } from 'react';
import { twMerge } from 'tailwind-merge';
import GridMarkupWrapper from './GridMarkupWrapper';

interface ColspanningGridMarkupProps extends ComponentPropsWithoutRef<'div'> {
    layoutConfig: 'col-span-2';
}

const ColspanningGridMarkup: FC<ColspanningGridMarkupProps> = ({ layoutConfig, className }) => {
    return (
        <GridMarkupWrapper
            markupComponent={<ColumnsSpanningGrid layoutConfig={layoutConfig} />}
            childrenTagsArr={colSpanningGridItemTagsToText}
            className={twMerge(className)} // Pass the className to the wrapper
        />
    );
};

export default ColspanningGridMarkup;
