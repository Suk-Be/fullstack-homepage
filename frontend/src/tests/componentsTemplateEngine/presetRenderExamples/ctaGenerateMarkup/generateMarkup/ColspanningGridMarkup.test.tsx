import ColspanningGridMarkup from '@/componentsTemplateEngine/presetRenderExamples/ctaGenerateMarkup/generateMarkup/ColspanningGridMarkup';
import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

describe('ColspanningGridMarkup', () => {
    it('renders the expected markup text', () => {
        render(<ColspanningGridMarkup layoutConfig="col-span-2" />);
        const styledIOpeningCodeBlock = screen.getByText(/col-spannning-grid-container/);
        const styledICodeBlockItems = screen.getAllByText(
            (content) =>
                content.includes('h-24 bg-green') || content.includes('h-24 bg-green-dark'),
        );

        expect(styledIOpeningCodeBlock).toBeInTheDocument();
        expect(styledICodeBlockItems).toHaveLength(6);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<ColspanningGridMarkup layoutConfig="col-span-2" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
