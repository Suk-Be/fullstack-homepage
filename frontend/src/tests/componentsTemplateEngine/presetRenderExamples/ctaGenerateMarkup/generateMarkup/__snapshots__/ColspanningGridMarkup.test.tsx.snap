// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColspanningGridMarkup > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-1 p-4"
    data-testid="grid-markup-wrapper"
  >
    <div
      class="bg-gray-dark text-green p-6 rounded-xl"
    >
      <code
        class="block pb-4"
        data-testid="opening-tag"
      >
        &lt;div class="grid grid-cols-3 xl:grid-cols-3 gap-4 px-4 py-4" data-testid="col-spannning-grid-container"&gt;
      </code>
      <code
        class="block pl-4"
        data-testid="children-tags-container"
      >
        <div>
          &lt;div class="w-full rounded-xl h-24 bg-green-dark" data-testid="grid-element"&gt;&lt;/div&gt;
        </div>
        <div>
          &lt;div class="w-full rounded-xl h-24 bg-green col-span-2" data-testid="grid-element"&gt;&lt;/div&gt;
        </div>
        <div>
          &lt;div class="w-full rounded-xl h-24 bg-green col-span-2" data-testid="grid-element"&gt;&lt;/div&gt;
        </div>
        <div>
          &lt;div class="w-full rounded-xl h-24 bg-green-dark" data-testid="grid-element"&gt;&lt;/div&gt;
        </div>
        <div>
          &lt;div class="w-full rounded-xl h-24 bg-green-dark" data-testid="grid-element"&gt;&lt;/div&gt;
        </div>
        <div>
          &lt;div class="w-full rounded-xl h-24 bg-green col-span-2" data-testid="grid-element"&gt;&lt;/div&gt;
        </div>
      </code>
      <code
        class="block pt-4"
        data-testid="closing-tag"
      >
        &lt;/div&gt;
      </code>
    </div>
  </div>
</DocumentFragment>
`;
